lsp-clients.el

```diff
@@ -179,13 +171,18 @@ finding the executable with variable `exec-path'."
 
 (lsp-register-client
  (make-lsp-client :new-connection (lsp-stdio-connection (lambda ()
-                                                          (cons lsp-clients-javascript-typescript-server
+                                                          (cons (lsp-package-path :npm "javascript-typescript-langserver" ".bin/javascript-typescript-stdio")
                                                                 lsp-clients-typescript-javascript-server-args)))
                   :activation-fn 'lsp-typescript-javascript-tsx-jsx-activate-p
                   :priority -3
                   :completion-in-comments? t
-                  :ignore-messages '("readFile .*? requested by TypeScript but content not available")
-                  :server-id 'jsts-ls))
+                  :server-id 'jsts-ls
+                  :download-server-fn (lambda (_client callback _update?)
+                                        (lsp-package-ensure
+                                         :npm
+                                         "javascript-typescript-langserver"
+                                         (lambda (success? msg)
+                                           (funcall callback success? msg))))))
```

Comment:
Here each callback function takes a boolean status flag as an argument. In JavaScript, promises take success and error callbacks separately. I think this is a matter of preference, but I'd personally prefer `lsp-package-ensure` having two distinct callback functions. What do you think? Depending on your decision, I will extend the API of my `nix-env-install` to support callbacks.

Reply:
I will experiment which interface result in better code.

Reply:
You might want to take a look at [emacs-async](https://github.com/jwiegley/emacs-async) for design of async APIs. It would be possible to support concurrent downloads, even if you probably don't need it in this package.